pkg/cloudmap/enqueue_requests_for_pod_events_test.go

```diff
@@ -0,0 +1,358 @@
+package cloudmap
+
+import (
+	"context"
+	appmesh "github.com/aws/aws-app-mesh-controller-for-k8s/apis/appmesh/v1beta2"
+	"github.com/aws/aws-app-mesh-controller-for-k8s/pkg/k8s"
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/util/workqueue"
+	testclient "sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"testing"
+)
+
+func Test_enqueueRequestsForPodEvents_Create(t *testing.T) {
+	vn1 := &appmesh.VirtualNode{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "global",
+			Name:      "vn-1",
+		},
+		Spec: appmesh.VirtualNodeSpec{
+			PodSelector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"app": "testapp",
+				},
+			},
+		},
+	}
+
+	type env struct {
+		virtualNodes []*appmesh.VirtualNode
+	}
+	type args struct {
```

Comment:
let's have a dedicated test case for enqueueVirtualNodesForPods, which have check on pod's selector == nil case.

Reply:
ok